core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -82,6 +82,7 @@ object Defaults {
   val MetadataSnapshotMaxNewRecordBytes = 20 * 1024 * 1024
   val ProducerBatchDecompressionEnable = true
   val PreferredController = false
+  val LiMinPreferredControllerCount = 0
```

Comment:
I see. So by default the cluster is treated as non-preferred-controller cluster. We will need to explicitly set this config value to enable the protection.

Reply:
Yes that's the intent.